monai/losses/ssim_loss.py

```diff
@@ -11,12 +11,12 @@
 
 import torch
 import torch.nn.functional as F
-from torch import nn
+from torch.nn.modules.loss import _Loss
 
 from monai.utils.type_conversion import convert_to_dst_type
 
 
-class SSIMLoss(nn.Module):
+class SSIMLoss(_Loss):
```

Comment:
the `forward` is currently hard-coded to return the mean value which is not compatible with the metrics... could you help add the `reduction` option to this class?

Reply:
[a2a351db00b91a3bd229a1c546dec48d5d76466f](https://github.com/Project-MONAI/MONAI/pull/5502/commits/a2a351db00b91a3bd229a1c546dec48d5d76466f)

Yes, I think metric should report for all elements in each batch, but SSIMLoss return a single value across in each batch. That was an issue.

Temporarily, I used for-loop and if-else, in the batch, but in long term, I think we should break the dependency between SSIMLoss and SSIM Metric, which is also true in DiceLoss and DiceMetric.

Reply:
Oh, so, rather than editing SSIMLoss,  I edited SSIMMetric, since current SSIMLoss makes sense for loss function.

Reply:
makes sense, thanks!